lib/service/desktop.go

```diff
@@ -248,7 +250,29 @@ func (process *TeleportProcess) initWindowsDesktopServiceRegistered(log *logrus.
 			log.Infof("Starting Windows desktop service on %v.", listener.Addr())
 		}
 		process.BroadcastEvent(Event{Name: WindowsDesktopReady, Payload: nil})
-		err := srv.Serve(listener)
+
+		muxCAGetter := func(ctx context.Context, id types.CertAuthID, loadKeys bool) (types.CertAuthority, error) {
+			return accessPoint.GetCertAuthority(ctx, id, loadKeys)
+		}
+
+		mux, err := multiplexer.New(multiplexer.Config{
+			Listener:                    listener,
+			EnableExternalProxyProtocol: cfg.Proxy.EnableProxyProtocol,
+			ID:                          teleport.Component(teleport.ComponentWindowsDesktop),
+			CertAuthorityGetter:         muxCAGetter,
```

Comment:
```suggestion
		mux, err := multiplexer.New(multiplexer.Config{
			Listener:                    listener,
			EnableExternalProxyProtocol: cfg.Proxy.EnableProxyProtocol,
			ID:                          teleport.Component(teleport.ComponentWindowsDesktop),
			CertAuthorityGetter:         accessPoint.GetCertAuthority,
```

Isn't this equivalent?

Reply:
Changed.